### PR TITLE
Problem: (CE-135) Missing addWitness method in transfer transaction builder

### PR DESCRIPTION
**Pull request by calvinlauco**
Solution: Add addWitness method

### CODE REVIEW

**Comment by calvinlauco** [comment]
@foreseaz This is a preview of what `addWitness` will be like. It accepts a `TxInWitness`.

File: lib/src/transaction/transfer/transfer_transaction_builder.spec.ts [hunk 1]
```diff
@@ -705,6 +707,157 @@ describe('TransferTransactionBuilder', () => {
...
+                .addViewKey(
+                    Buffer.from(
+                        '0248b7c5f2325a7ef7dcd68066368fd63a7aad8c4a894414fcd81b227b2178322c',
+                        'hex',
+                    ),
+                );
+            const anotherKeyPair = KeyPair.generateRandom();
+            const witness = native.transferTransaction.schnorrSignMessage(
+                Buffer.from(builder.txId(), 'hex'),
+                anotherKeyPair.toObject(),
```
Comment (foreseaz) [inline lib/src/transaction/transfer/transfer_transaction_builder.spec.ts:812]: I may need more context here, so the witness should be the `keypair` of TransferAddress stated in Input? If so, the Transfer wallet is to sign the whole tx, why add the witness again?

File: lib/src/transaction/transfer/transfer_transaction_builder.ts [hunk 2]
```diff
@@ -283,7 +279,44 @@ export class TransferTransactionBuilder {
...
+        } else {
+            throw new Error(
+                `Unsupported fee algorithm ${this.feeConfig.algorithm}`,
+            );
+        }
+
+        this.incompleteHex = updatedIncompleteSigningHex;
+    }
+
+    private owIndex(): NumberPredicate {
```
Comment (foreseaz) [inline lib/src/transaction/transfer/transfer_transaction_builder.ts:318]: neat ow checking

File: lib/src/transaction/transfer/transfer_transaction_builder.spec.ts [hunk 3]
```diff
@@ -705,6 +707,157 @@ describe('TransferTransactionBuilder', () => {
...
+                    value: new BigNumber('1000'),
+                })
+                .addViewKey(
+                    Buffer.from(
+                        '0248b7c5f2325a7ef7dcd68066368fd63a7aad8c4a894414fcd81b227b2178322c',
+                        'hex',
+                    ),
+                );
+            const witness = Buffer.from(
+                '001e649e6ec3165c63b0fa47ffce7d18675f49bf887354a21d0f8ecc4d2be6cb784aad962752d1550913cefdfd7211f97bb80af5713a4692f41ab20bd88c0bc940ebaac52a25455a63e812b57a42439c68c016d05f7d99e84561a2728aebe872a30002fe0a593b63d48ee042035f39432d062c5df78876d16b9c328044bd6e120b7392',
```
[the remark this answers is not in the document]
Comment (calvinlauco) [reply lib/src/transaction/transfer/transfer_transaction_builder.spec.ts:737]: The witness will come from combining `SchnorrSignature` obtained from `MultiSigSession` and the `MerkleTree` proof of the signing public keys.

You can find a code example here:
https://github.com/crypto-com/chain/blob/release/v0.3/chain-core/src/tx/mod.rs#L330

File: lib/src/transaction/transfer/transfer_transaction_builder.spec.ts (hunk 1 above)
Comment (calvinlauco) [reply lib/src/transaction/transfer/transfer_transaction_builder.spec.ts:812]: This test case is to try to generate the signature from Rust code and input it to `addWitness` to make sure the  method works as expected. For `MultiSig` the witness will be generated from the `MultiSigSession`.

**Comment by calvinaco** [comment]
bors try

**Pull request merged**